documentation/SolutionsRegistryFormat.md

```diff
@@ -0,0 +1,196 @@
+
+**Example blocks**:
+```
+"configure": [
+    "settings.myconf"
+],
+"restore": [
+    "settings.myconf"
+],
+"start": [
+    {
+        "type": "gpii.launch.exec",
+        "command": "\"${{registry}.HKEY_LOCAL_MACHINE\\SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\App Paths\\nvda.exe\\}\""
+    }
+],
+"stop": [
+    {
+        "type": "gpii.windows.killProcessByName",
+        "filename": "nvda.exe"
+    }
+]
+```
+
+### update
+
+The `update` block works very similarly to the configure, restore, start and stop blocks. It describes what should happen when the configuration needs to be updated (eg. due to context changes, PCP adjustments, etc).
+
+The format of the `update` block allows for the same entries as the configure, restore, start and stop blocks - that is: arbitrary lifecycle action blocks and `settings.<blockname>`. Unlike for the other lifecycle blocks, the `update` block furthermore allows references to the `start`, `stop` and `configure` blocks. This is one by putting a string with the name of that block. When the system encounters one of these references, the entries of that block will be run.
+
+**Example block**:
+```
+"configure": [
+    "settings.myconf"
+],
+"update": [
+    "configure",
+    {
+        "type": "gpii.launch.exec",
+        "command": "my_application --refresh"
+    }
+]
+```
+
+In the above example, the process of updating the application settings would consists of running the contents of the `configure` block (that is `"settings.myconf"`), followed by a custom lifecycle actions.
+
+
+### isInstalled:
+
+This directive is used to detect whether a solution is installed. If any of these blocks evaluate to `true` (implicit **OR**), the application is considered to be installed.
+
+**Example Entry**:
+```
+"isInstalled": [
+    {
+        "type": "gpii.reporter.fileExists",
+        "fileName": "${{registry}.HKEY_CURRENT_USER\\Software\\Texthelp\\Read&Write10\\InstallPath}\\ReadAndWrite.exe"
+    }, // SEEMS LIKE IMPLICIT OR BETWEEN THESE BLOCKS
+    {
+        "type": "gpii.packageKit.find",
+        "name": "orca"
+    }
+]
+```
+
+*****
+
+### UNIMPLEMENTED BLOCKS
+There are several advanced options that we're not planning to implement in the short term, but which will make the implementation of things like the ORCA settings handler much less horrible.
+
+#### isRunning:
+To detect whether a solution is running - this is planned to be integrated in the relatively short run. If any of these blocks evaluate to `true` (implicit OR), the application is considered to be running.
+
+**Example Entry**:
+```
+"isRunning": [
+    {
+        "type": "gpii.processReporter.byName",
+        "fileName": "firefox.exe"
+    },
+    {
+       "type": "gpii.processReporter.find",
+       "command": "ps aux | grep 'myprocess | wc -l'",
```

Comment:
@klown This documentation was actually just meant as examples of how things might look once we get the proper support for process detection - so it was meant to follow your format.. Not sure where I got my examples from :) ... Anyway, I've updated it to be similar to your example from your code... I imagine that we might want to have some more detailed documentation about the available process reporters which we would refer to from the solutions registry documentation.. Anyway - hopefully it looks correct now. (PS: I've removed the second block, cause I couldn't really think up a good sensible example for it right now)

@amb26 I think @klown was just referring to my horrific syntax and linux command line skills.. I think we in all cases so far can detect whether a solution is running by a single directive (or one of multiple single directives)... But I might have missed something